src/linker.cpp

```diff
@@ -631,7 +627,7 @@ gb_internal i32 linker_stage(LinkerData *gen) {
 					link_settings = gb_string_append_fmt(link_settings, "-mmacosx-version-min=%.*s ", LIT(build_context.minimum_os_version_string));
 				}
 
-				if (build_context.build_mode != BuildMode_DynamicLibrary) {
+				if (build_context.build_mode != BuildMode_DynamicLibrary && build_context.build_mode != BuildMode_StaticLibrary) {
```

Comment:
This does not look needed? `link_settings` isn't used when you do a static library from what I can tell.